include/PreCharge/GFDB.hpp

```diff
@@ -0,0 +1,125 @@
+
+
+#ifndef _EVT_GFDB_H
+#define _EVT_GFDB_H
+
+#include <EVT/io/CAN.hpp>
+#include <EVT/utils/time.hpp>
+#include <stddef.h>
+
+namespace IO = EVT::core::IO;
+
+namespace GFDB {
+
+/**
+ * SIM100 Driver for ground fault detection
+ * https://sendyne.com/Datasheets/Sendyne%20SIM100MOD%20Datasheet%20V1.5.pdf
+ * https://sendyne.com/Datasheets/SIM100_CAN_protocol_v0.8a.pdf
+ */
+class GFDB {
+public:
+    GFDB(IO::CAN& can);
+
+    /**
+     * Connects the GFDB's CAN to the network
+     * @return CAN Connection status
+     */
+    IO::CAN::CANStatus init();
```

Comment:
Now that I think about it, `CAN.connect` should only be run once, and it should probably be done directly in main. It might make more sense to just delete this method entirely